src/browser/web-view.js

```diff
@@ -393,23 +398,49 @@ define((require, exports, module) => {
       width: '100vw',
       height: 'calc(100vh - 28px)',
     },
-    active: {
-      transition: 'opacity 150ms linear',
+    fadeIn: {
+      transition: 'opacity 100ms linear',
+      transform: 'scale(1)',
+      opacity: 1,
     },
-    inactive: {
-      transition: 'transform 300ms linear, opacity 150ms linear',
+    fadeOut: {
+      transition: 'transform 0ms linear 100ms, opacity 100ms linear',
+      transform: 'scale(0)',
+      opacity: 0,
+      pointerEvents: 'none',
+    },
+    grow: {
+      transition: 'transform 200ms linear, opacity 200ms linear',
+      transform: 'scale(1)',
+      opacity: 1,
+    },
+    shrink: {
+      transition: 'transform 200ms linear, opacity 150ms linear',
       transform: 'scale(0)',
       opacity: 0,
       pointerEvents: 'none',
     }
   });
 
-  const view = (mode, loader, shell, page, address, selected) => {
-    const isActive = mode === 'show-web-view';
+  const view = (mode, transition, loader, shell, page, address, selected) => {
+    const additionalStyles =
```

Comment:
Can we have something like `Style(webviewsStyle.base, webviewsStyle[getModeStyle(mode, transition)])` instead of inlining all this logic.